contracts/ClaimZap.sol

```diff
@@ -0,0 +1,199 @@
+        address _locker
+    ) {
+        veAsset = _veAsset;
+        veToken = _veToken;
+        ve3Token = _ve3Token;
+        veAssetDeposit = _veAssetDeposit;
+        ve3TokenRewards = _ve3TokenRewards;
+        ve3dRewards = _ve3dRewards;
+        exchange = _exchange;
+        ve3dLocker = _locker;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "ClaimZap V2.0";
+    }
+
+    function setApprovals() external onlyOwner{
+        IERC20(veAsset).safeApprove(veAssetDeposit, 0);
+        IERC20(veAsset).safeApprove(veAssetDeposit, type(uint256).max);
+        IERC20(veAsset).safeApprove(exchange, 0);
+        IERC20(veAsset).safeApprove(exchange, type(uint256).max);
+
+        IERC20(veToken).safeApprove(ve3dRewards, 0);
+        IERC20(veToken).safeApprove(ve3dRewards, type(uint256).max);
+
+        IERC20(ve3Token).safeApprove(ve3TokenRewards, 0);
+        IERC20(ve3Token).safeApprove(ve3TokenRewards, type(uint256).max);
+
+        IERC20(veToken).safeApprove(ve3dLocker, 0);
+        IERC20(veToken).safeApprove(ve3dLocker, type(uint256).max);
+    }
+
+    function CheckOption(uint256 _mask, uint256 _flag) internal pure returns(bool){
+        return (_mask & (1<<_flag)) != 0;
+    }
+
+    /// @notice Claim rewards from multiple reward pools in one transaction and allow passing options for staking, claiming locked balances etc.
+    /// @param rewardContracts Array of contract addresses to claim rewards from
+    /// @param extraRewardContracts Array of contract addresses to claim extra rewards from
+    /// @param tokenRewardContracts Array of contract addresses to claim multiple reward tokens from
+    /// @param tokenRewardTokens Array of token addresses to claim in multiple token reward contracts
+    /// @param depositVeAssetMaxAmount Maximum amount of VeAsset to lock and stake
+    /// @param minAmountOut Minimum amount to swap in exchange
+    /// @param depositVeTokenMaxAmount Maximum amount of VeToken to stake
+    /// @param options number that represents the options for claiming extras
+    function claimRewards(
+        address[] calldata rewardContracts,
+        address[] calldata extraRewardContracts,
+        address[] calldata tokenRewardContracts,
+        address[] calldata tokenRewardTokens,
+        uint256 depositVeAssetMaxAmount,
+        uint256 minAmountOut,
+        uint256 depositVeTokenMaxAmount,
+        uint256 options
+    ) external{
+        uint256 veAssetBalance = IERC20(veAsset).balanceOf(msg.sender);
+        uint256 veTokenBalance = IERC20(veToken).balanceOf(msg.sender);
+
+        //claim from main LP pools
+        for(uint256 i = 0; i < rewardContracts.length; i++){
+            IBasicRewards(rewardContracts[i]).getReward(msg.sender,true);
+        }
+        //claim from extra rewards
+        for(uint256 i = 0; i < extraRewardContracts.length; i++){
+            IBasicRewards(extraRewardContracts[i]).getReward(msg.sender);
+        }
```

Comment:
the extra reward pool is claimed in the same function get reward of the parent reward pool ,so you don't need to claim it by a separate txs , or this related to smth else

Reply:
These 3 arrays of contracts are for LP pools, extra reward contracts and multi reward token contracts. They each have another interface for getting rewards so you can put in the contract addresses in those arrays depending on which interface is used. If you are certain we claim all these contracts from their parent then we may not need it, though maybe it could be good to keep the flexibility for future claims

Reply:
yes all lp reward pools are instance of base reward pool and in its get reward function includes the get extra reward pools and you already pass the second parameter as true ,so it will collect the extra rewards, so 
https://github.com/veTokenFinance/veToken-contracts/blob/main/contracts/BaseRewardPool.sol#L302

is there is a case that user want to claim from extra pools without the main pool? , doesn't make sense for me
@jetbrain10